common/src/main/java/com/wynntils/services/itemfilter/filters/PercentageStatFilter.java

```diff
@@ -33,6 +37,39 @@ protected boolean matches(StatValue value) {
         return percentage >= min && percentage <= max;
     }
 
+    @Override
+    public String asString() {
+        if (min == max) {
+            return StringUtils.floatToSimpleString(min) + "%";
+        }
+
+        if (min == Integer.MIN_VALUE) {
+            return equalsInString
+                    ? "<=" + StringUtils.floatToSimpleString(min)
+                    : "<" + StringUtils.floatToSimpleString(max + 1);
+        }
+
+        if (max == Integer.MAX_VALUE) {
+            return equalsInString
+                    ? ">=" + StringUtils.floatToSimpleString(min)
+                    : ">" + StringUtils.floatToSimpleString(min - 1);
+        }
+
+        return StringUtils.floatToSimpleString(min) + "-" + StringUtils.floatToSimpleString(max) + "%";
```

Comment:
The problem with this is that the greater/less than isn't going into the correct if, it's returning the ranged format

Reply:
How so?

Reply:
In the screenshot example you would expect the `asString` to go into
```
if (max == Integer.MAX_VALUE) {
    return equalsInString
        ? ">=" + StringUtils.floatToSimpleString(min)
        : ">" + StringUtils.floatToSimpleString(min - 1);
}
```
but it isn't and it's returning

```StringUtils.floatToSimpleString(min) + "-" + StringUtils.floatToSimpleString(max) + "%";``` and since there's no valid max value it's returning the weird output

Reply:
Because it should be Float.MaxValue... My bad

Reply:
And it also needs the "%" added to the end, just noticed that